src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -105,6 +110,7 @@ public BinaryJedisCluster(Set<HostAndPort> jedisClusterNode, int connectionTimeo
     this.connectionHandler = new JedisSlotBasedConnectionHandler(jedisClusterNode, poolConfig,
         connectionTimeout, soTimeout, infiniteSoTimeout, user, password, clientName, ssl, sslSocketFactory, sslParameters, hostnameVerifier, hostAndPortMap);
     this.maxAttempts = maxAttempts;
+    this.timeout = Duration.ofMillis(Math.max(connectionTimeout, soTimeout));
```

Comment:
1. We already have three different timeouts. Use a better name than `timeout` so that it is easily understandable.
2. There should be an option to set this timeout. It's fine to add another parameter in constructor (in a new constructor of course).
3. `connectionTimeout` should not be a part of this. `soTimeout` can be used as default.

Reply:
Done!